<a href="https://colab.research.google.com/github/vitorjensen/Python_para_Analise_de_Dados/blob/main/Projetos/An%C3%A1lise%20Explorat%C3%B3ria%20de%20Combust%C3%ADvel/An%C3%A1lise_Explorat%C3%B3ria_de_Combust%C3%ADvel.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Importando bibliotecas necessárias:

In [3]:
import yfinance as yf
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import numpy as np
from datetime import datetime

In [ ]:
# pip install yfinance pandas

**Objetivo:** Fazer uma análise exploratória do Brent (Petróleo bruto) no período de 01/2022 a 09/2026

Primeira parte: Extração da base de dados utilizando o Yahoo Finance.

In [ ]:
# Definindo o ticker (Brent) a ser manipulado
ticker = "BZ=F"

# Criando o dataset com o período de dados estabelecido
data_inicio = "2022-01-01"
data_fim = datetime.now().strftime('%Y-%m-%d')

# Definindo um objeto para o ticker
brent = yf.Ticker(ticker)

# Criando o dataframe com os dados do Yahoo extraídos
df_brent = brent.history(start=data_inicio, end=data_fim)

display(df_brent.head())

In [ ]:
# Fazendo download em CSV da base para a máquina

df_brent = df_brent.to_csv('dados_brent.csv')

Segunda parte: Tratamento.

- Excluir colunas indesejadas
- Renomear colunas
- Definir Total de colunas e linhas.
- Verificar valores zeros e duplicados
- Ajustar tipo do campo Data
- Verificar se há datas duplicadas

In [ ]:
# Lendo aquivo CSV
df = pd.read_csv('/content/drive/MyDrive/dados_brent.csv')

# Total de colunas e linhas
display(df)

In [ ]:
# informações gerais do dataset

df.info()

In [ ]:
# Excluindo colunas indesejadas:
colunas_indesejadas = ['Dividends', 'Stock Splits']
df = df.drop(colunas_indesejadas, axis=1, errors='ignore')

display(df)

In [6]:
# Total de linhas e colunas
print(f'Total de linhas: {df.shape[0]}')
print(f'Total de colunas: {df.shape[1]}')

Total de linhas: 1190
Total de colunas: 6


In [ ]:
# Renomear colunas
df = df.rename(columns={
    'Date': 'Data',
    'Open': 'Valor_Abertura',
    'Low': 'Valor_Minimo',
    'High': 'Valor_Maximo',
    'Close': 'Fechamento'
})

display(df)

In [ ]:
# Verificando existência de valores "zero"

geral_zeros = df[(df == 0).any(axis=1)]
print(f"Quantidade de linhas contendo pelo menos um zero: {len(geral_zeros)}")
display(geral_zeros)

In [14]:
# Verificando total de duplicados

duplicados = df.duplicated().sum()
print(f'Total de valores duplicados: {duplicados}')

Total de valores duplicados: 0


Resumo do tratamento:

- Total de colunas removidas: 2 (Dividends e Stock Split)
- Total de linhas e colunas: 1190 x 6
- Total de colunas renomeadas: 5
- Total de ocorrências de valores 0: 8
- Total de valores duplicados: 0

Terceira parte da Análise Exploratória:

- Colunas categóricas.
- Colunas numéricas e sua Análise Univariada.
- Correlação Linear
- Outras análises pertinentes ao conjunto de dados.

In [ ]:
# Verificando colunas categóricas:

colunas_categoricas =[columns for columns in df.columns
                      if df[columns].dtype == 'object']

qtde_total = len(colunas_categoricas)

print(f'Qtde de colunas categóricas: {qtde_total}')

print(f'Colunas categóricas: {colunas_categoricas}')

In [ ]:
# Verificando colunas numéricas

colunas_numericas = [columns for columns in df.columns
                     if df[columns].dtype != 'object']

qtde_total = len(colunas_numericas)

print(f'Qtde de colunas numéricas: {qtde_total}')
print(f'Colunas numéricas: {colunas_numericas}')

